ModeratorSignUpDone Inshallah

aabdoo23 opened the pull request.

aabdoo23 committed:
ModeratorSignUpDone Inshallah

MostafaHgz commented on SchedsForums.Application/Commands/Moderators/SignUp/ModeratorSignUpCommandHandler.cs:
```diff
@@ -0,0 +1,39 @@
+        IPasswordService passwordService)
+        : IRequestHandler<ModeratorSignUpCommand, ModeratorSignUpResponseDTO>
+    {
+        private readonly IBaseRepository<ModeratorSignUpRequest> _moderatorSignUpRequestRepository = moderatorSignUpRequestRepository;
```
add null check

MostafaHgz commented on SchedsForums.Application/Commands/Moderators/SignUp/ModeratorSignUpCommandHandler.cs:
```diff
@@ -0,0 +1,39 @@
+        private readonly IBaseRepository<ModeratorSignUpRequest> _moderatorSignUpRequestRepository = moderatorSignUpRequestRepository;
+        private readonly IPasswordService _passwordService = passwordService;
+
+        public async Task<ModeratorSignUpResponseDTO> Handle(ModeratorSignUpCommand request, CancellationToken cancellationToken)
```
rename to command

MostafaHgz commented on SchedsForums.Application/Commands/Moderators/SignUp/ModeratorSignUpResponseDTO.cs:
```diff
@@ -0,0 +1,8 @@
+namespace SchedsForums.Application.Commands.Moderators.SignUp
+{
+    public class ModeratorSignUpResponseDTO: UserResponseDTO
+    {
```
add reason

MostafaHgz commented on SchedsForums.Application/SchedsForums.Application.csproj:
```diff
@@ -24,7 +24,7 @@
 
   <ItemGroup>
-    <Folder Include="Queries\Admins\" />
+    <Folder Include="Queries\Admins\GetAllModeratorSignUpRequests\" />
```
until admin

MostafaHgz commented on SchedsForums.Domain/Entities/ModeratorSignUpRequest.cs:
```diff
@@ -0,0 +1,15 @@
+        public string Email { get; set; }
+        public string PasswordHash { get; set; }
+        public string Reason { get; set; }
+        public bool IsApproved { get; set; }
```
make this a status enum

MostafaHgz replied:
add status updated by

MostafaHgz commented on SchedsForums.Domain/Entities/ModeratorSignUpRequest.cs:
```diff
@@ -0,0 +1,15 @@
+        public string PasswordHash { get; set; }
+        public string Reason { get; set; }
+        public bool IsApproved { get; set; }
+        public DateTime? ApprovedAt { get; set; }
```
status updatedAt

MostafaHgz commented on SchedsForums.Infrastructure/ServiceCollectionExtensions.cs:
```diff
@@ -47,26 +50,31 @@ public static IServiceCollection AddInfrastructureServices(this IServiceCollecti
             services.Configure<JwtOptions>(options =>
             {
-                configuration.GetSection("Jwt").Bind(options);
                 options.Key = Environment.GetEnvironmentVariable("JWT_KEY")
```
move to constants

MostafaHgz commented on SchedsForums.Domain/Entities/Users/Moderator.cs:
```diff
@@ -0,0 +1,10 @@
+
+namespace SchedsForums.Domain.Entities.Users
+{
+    public class Moderator : BaseUser
```
use pending moderator and moderator

aabdoo23 committed:
Refactor ModeratorSignUp to PendingModerator

Replaced ModeratorSignUpRequest with PendingModerator entity across the codebase. Updated related classes, including ModeratorSignUpCommandHandler, ModeratorSignUpResponseDTO, and Moderator. Modified SchedsForumsDbContext to reflect the new entity and updated the database schema accordingly. Added JWTConstants class for JWT configuration keys. Added new migration for schema changes.

MostafaHgz commented on SchedsForums.Application/Commands/Moderators/SignUp/ModeratorSignUpCommandHandler.cs:
```diff
@@ -0,0 +1,45 @@
+    {
+        private readonly IBaseRepository<PendingModerator> _moderatorSignUpRequestRepository = moderatorSignUpRequestRepository 
+            ?? throw new ArgumentNullException(nameof(moderatorSignUpRequestRepository));
+        private readonly IPasswordService _passwordService = passwordService;
```
add null check

MostafaHgz commented on SchedsForums.Application/Commands/Moderators/SignUp/ModeratorSignUpCommandHandler.cs:
```diff
@@ -0,0 +1,45 @@
+        {
+            var hashedPassword = _passwordService.HashPassword(command.Password);
+
+            var PendingModerator = new PendingModerator
```
use camelCase

MostafaHgz commented on SchedsForums.Application/Commands/Moderators/SignUp/ModeratorSignUpCommandHandler.cs:
```diff
@@ -0,0 +1,45 @@
+
+            //is this correct? reassigning the variable and using that to return, or is it redundant
+            //i did it for the id mainly, but now i'm not sure
+            PendingModerator = await _moderatorSignUpRequestRepository.InsertAsync(PendingModerator);
```
no need to reassign just the insert will add the id to the object

MostafaHgz commented on SchedsForums.Application/SchedsForums.Application.csproj:
```diff
@@ -23,12 +23,12 @@
 
-  <ItemGroup>
-    <Folder Include="Queries\Admins\" />
+  <!--<ItemGroup>-->
```
if it's working while commented then remove it

MostafaHgz commented on SchedsForums.Domain/Entities/Users/PendingModerator.cs:
```diff
@@ -0,0 +1,18 @@
+        public virtual Admin? StatusUpdatedBy { get; set; }
+        public ModeratorStatus Status { get; set; }
+    }
+    public enum ModeratorStatus
```
move this to a separate file

MostafaHgz commented on SchedsForums.Infrastructure/ConfigurationOptions/JWTConstants.cs:
```diff
@@ -0,0 +1,10 @@
+    {
+        public static string JWT_KEY = "JWT_KEY";
+        public static string JWT_Options = "JwtOptions";
+        public static string JWT_ISSUER = JWT_Options+":Issuer";
```
do string interpolation using $""

MostafaHgz commented on SchedsForums.Infrastructure/Migrations/20241101080040_PendingModerator.cs:
```diff
@@ -0,0 +1,166 @@
+
+#nullable disable
+
+namespace SchedsForums.Infrastructure.Migrations
```
as I told you before down this migration and the one before it and make this feature in only one migration as there some inconsistencies in this migration

MostafaHgz commented on SchedsForums.Infrastructure/Migrations/20241101080040_PendingModerator.cs:
```diff
@@ -0,0 +1,166 @@
+                name: "ApprovedById",
+                table: "Users");
+
+            migrationBuilder.RenameColumn(
```
like this one

aabdoo23 committed:
Refactor and update Moderator and PendingModerator logic

- Added null check and corrected variable name in ModeratorSignUpCommandHandler.cs
- Removed commented-out folder includes in SchedsForums.Application.csproj
- Removed ModeratorStatus enum from PendingModerator.cs
- Updated JWTConstants.cs to use string interpolation
- Removed old migration files: 20241029114409_login, 20241031100520_ModeratorMigration, 20241101080040_PendingModerator
- Added ModeratorStatus.cs to SchedsForums.Domain.Entities.Users
- Added new migration 20241101132322_PendingModerator to create Users table with updated columns and relationships
